_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -85,6 +85,87 @@ bool CheckHardwareSupport(VideoCORE *p_core, mfxVideoParam *p_video_param)
     return true;
 }
 
+class ReferenceFrameStorage
+{
+private:
+    std::vector<UMC_VP9_DECODER::VP9DecoderFrame> m_submittedFrames;
```

Comment:
As for  me, I don't see any problems w/ `std::find` since number of elements is small enough. At the same time, having same mem ID both as a map's key and as member of `VP9DecoderFrame` is redundant and confusing. I'd rather use `vector` or `set` at least.